api/handler/plan_api_test.go

```diff
@@ -0,0 +1,101 @@
+//go:build integration
+
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/compliance-framework/configuration-service/domain"
+	"github.com/compliance-framework/configuration-service/event/bus"
+	"github.com/compliance-framework/configuration-service/service"
+	"github.com/compliance-framework/configuration-service/tests"
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+)
+
+func TestPlanApi(t *testing.T) {
+	suite.Run(t, new(PlanIntegrationSuite))
+}
+
+type PlanIntegrationSuite struct {
+	tests.IntegrationTestSuite
+}
+
+func (suite *PlanIntegrationSuite) TestCreatePlan() {
+	suite.Run("A plan can be created through the API", func() {
+		e := echo.New()
+		logger, _ := zap.NewProduction()
+		planHandler := NewPlanHandler(logger.Sugar(), service.NewPlanService(bus.Publish))
+
+		e.POST("/plan", planHandler.CreatePlan)
```

Comment:
This is a good idea for us to add the actual POST piece. Maybe we can just register the handlers routes using the server function, and creating a group called `api/plan` ?

Reply:
https://github.com/compliance-framework/configuration-service/pull/142/files#diff-63de9eb2ef299afee6ce75f0fc1155b74548ffda5759f7d57c3e0c7ef851fe97R20